internal/fipcontroller/fipcontroller.go

```diff
@@ -196,6 +198,22 @@ func (fc *Controller) Reconcile() {
 	})
 }
 
+func (fc *Controller) fipChanged(oldFIP *hcloud.FloatingIP, newFIP *hcloud.FloatingIP) bool {
+	if oldFIP.ID != newFIP.ID {
+		return true
+	}
+
+	if oldFIP.Server != newFIP.Server {
+		if oldFIP.Server != nil && newFIP.Server != nil && oldFIP.Server.ID == newFIP.Server.ID {
+			return false
+		}
+
+		return true
+	}
+
+	return false
```

Comment:
if we add the helper suggested above, this whole function could become 
```suggestion
	return oldFIP.ID != newFIP.ID || fipServerName(oldFIP) != fipServerName(newFIP) 
```

This should work because hcloud ensures server names are unique in a project.
It has the not-so-nice property of triggering a reconciliation on server renaming, but renamings would break a lot of other things as well (like k8s itself), so I think it's an acceptable simplification.

Reply:
That would require additional API calls, server name is empty for newFIP - server objects returned by `hcloudClient.FloatingIP().AllWithOpts` don't have it.

I'm fine with changing that if you think it's worth it regardless .

Reply:
ah yes, good point. Then I guess we'll have to live with the bigger function. Not the end of the world.